Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameFile.java

```diff
@@ -62,7 +66,16 @@ private GameFile(long pointer)
 
   public String getCoverPath(Context context)
   {
-    return context.getExternalCacheDir().getPath() + "/GameCovers/" + getGameTdbId() + ".png";
+    File externalCacheDir = context.getExternalCacheDir();
+
+    if (externalCacheDir != null)
+    {
+      return externalCacheDir.getPath() + "/GameCovers/" + getGameTdbId() + ".png";
+    }
+    else
+    {
+      throw new IllegalStateException("[GameFile] externalCacheDir is null");
```

Comment:
If this is null, we can't expect Dolphin to function at all. I don't think adding an explicit check for it here adds much.

Reply:
Is it possible that another app modifies `externalCacheDir`, which is `Nullable`? I know there are some questionable cleanup apps out there... I don't think the overhead of an if statement here is too atrocious but correct me if I'm misunderstanding something.

Reply:
> Is it possible that another app modifies `externalCacheDir`, which is `Nullable`? I know there are some questionable cleanup apps out there...

The case where this would be `null` is not when the directory has been deleted, it's when external storage is not mounted.

> I don't think the overhead of an if statement here is too atrocious but correct me if I'm misunderstanding something.

There's no problem performance-wise. I was thinking more about the maintenance side of things, if we should add null checks everywhere where Android Studio warns us even if the added checks don't actually solve anything.

Reply:
> The case where this would be null is not when the directory has been deleted, it's when external storage is not mounted.

It is possible for another app to unmount external storage. I believe the SAF can also do this, as there's an "eject" symbol next to the SD card on my phone.